packages/components/src/border-box-control/test/index.js

```diff
@@ -277,8 +288,22 @@ describe( 'BorderBoxControl', () => {
 			);
 
 			const colorButtons = screen.getAllByLabelText( colorPickerRegex );
+			expect( colorButtons ).toHaveLength( 4 );
+
+			for ( let i = 0; i < colorButtons.length; i++ ) {
+				// Click on the color button to show the color picker popover.
+				await user.click( colorButtons[ 0 ] );
+
+				// Wait for color picker popover to fully appear
+				const colorPickerButton = screen.getByRole( 'button', {
+					name: 'Custom color picker.',
+				} );
```

Comment:
I've noticed that queries themselves tend to be slow. But making `hidden: true` in order to disable additional work to exclude invisible elements makes this twice faster. 

I've ran this:

```
				console.time( 'without hidden' );
				const colorPickerButton = screen.getByRole( 'button', {
					name: 'Custom color picker.',
				} );
				console.timeEnd( 'without hidden' );

				console.time( 'with hidden' );
				screen.getByRole( 'button', {
					name: 'Custom color picker.',
					hidden: true,
				} );
				console.timeEnd( 'with hidden' );
```

and got these results:

```
  console.time
    without hidden: 89 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:304:13)

  console.time
    with hidden: 42 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:311:13)

  console.time
    without hidden: 98 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:304:13)

  console.time
    with hidden: 37 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:311:13)

  console.time
    without hidden: 70 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:304:13)

  console.time
    with hidden: 36 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:311:13)

  console.time
    without hidden: 71 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:304:13)

  console.time
    with hidden: 34 ms

      at Object.<anonymous> (packages/components/src/border-box-control/test/index.js:311:13)
```

So adding `hidden: true` for all queries here should also improve the situation.

That being said, I wonder if we should enable it globally:

https://testing-library.com/docs/dom-testing-library/api-configuration#defaulthidden

That's likely going to cause some issues in tests where we assume that the hidden elements will be excluded. But it might end up making the tests run faster. 

WDYT?

Reply:
Seems that anything that checks visibility, be it `getByRole` with `hidden: true` or the popover wait (but does it really check visibility of anything?) is ridiculously show. That's because there's plenty of `window.getComputedStyle()` calls, and each takes approx 30ms. That's a lot of time for something that traverses a tree and looks for object properties. In the profiler, flame chart view like this is typical:

<img width="597" alt="Screenshot 2022-12-19 at 8 18 28" src="https://user-images.githubusercontent.com/664258/208427629-42cefc9c-36d7-4b40-af83-e094671a797e.png">

We can "optimize" our tests to avoid visibility tests, but having to do that is really silly, the test author shouldn't worry about that at all.

Reply:
I agree. Ideally, the testing framework should be responsible for that, and since testing-library doesn't do that, we can enable it for the entire project as I linked above. While that's not perfect, it could at least solve the "the test author shouldn't worry about that at all." problem to some degree.